test/dsets.c

```diff
@@ -8788,203 +8788,220 @@ test_chunk_cache(hid_t fapl)
+    if ((dapl2 = H5Dget_access_plist(dsid)) < 0)
+        FAIL_STACK_ERROR;
+    if (H5Pget_chunk_cache(dapl2, &nslots_4, &nbytes_4, &w0_4) < 0)
+        FAIL_STACK_ERROR;
+    if ((nslots_2 != nslots_4) || (nbytes_2 != nbytes_4) || !H5_DBL_ABS_EQUAL(w0_2, w0_4))
+        FAIL_PUTS_ERROR("    Cache values from retrieved dapl do not match those from dapl2.");
+
+    /* Test H5D_CHUNK_CACHE_NSLOTS_DEFAULT and H5D_CHUNK_CACHE_W0_DEFAULT */
+    nslots_2 = H5D_CHUNK_CACHE_NSLOTS_DEFAULT;
+    w0_2     = H5D_CHUNK_CACHE_W0_DEFAULT;
     if (H5Pset_chunk_cache(dapl2, nslots_2, nbytes_2, w0_2) < 0)
-        FAIL_STACK_ERROR
+        FAIL_STACK_ERROR;
 
     if (H5Dclose(dsid) < 0)
-        FAIL_STACK_ERROR
+        FAIL_STACK_ERROR;
     if ((dsid = H5Dopen2(fid, "dset", dapl2)) < 0)
-        FAIL_STACK_ERROR
+        FAIL_STACK_ERROR;
+    if (H5Pclose(dapl2) < 0)
+        FAIL_STACK_ERROR; /* Close dapl2, to avoid id leak */
+    if ((dapl2 = H5Dget_access_plist(dsid)) < 0)
+        FAIL_STACK_ERROR;
+    if (H5Pget_chunk_cache(dapl2, &nslots_4, &nbytes_4, &w0_4) < 0)
+        FAIL_STACK_ERROR;
+    if ((nslots_3 != nslots_4) || (nbytes_2 != nbytes_4) || !H5_DBL_ABS_EQUAL(w0_3, w0_4))
+        FAIL_PUTS_ERROR("    Cache values from retrieved dapl do not match those expected.");
+    if (H5Pclose(dapl2) < 0)
+        FAIL_STACK_ERROR;
+
+    /* Verify that the file has indeed started using the new cache values (test
+     * use of H5Oopen with H5P_DEFAULT) */
+    if (H5Dclose(dsid) < 0)
+        FAIL_STACK_ERROR;
+    if ((dsid = H5Oopen(fid, "dset", H5P_DEFAULT)) < 0)
+        FAIL_STACK_ERROR;
+    if ((dapl2 = H5Dget_access_plist(dsid)) < 0)
+        FAIL_STACK_ERROR;
+    if (H5Pget_chunk_cache(dapl2, &nslots_4, &nbytes_4, &w0_4) < 0)
+        FAIL_STACK_ERROR;
+    if ((nslots_3 != nslots_4) || (nbytes_3 != nbytes_4) || !H5_DBL_ABS_EQUAL(w0_3, w0_4))
+        FAIL_PUTS_ERROR("    Cache values from retrieved dapl do not match those from fapl.");
+    if (H5Pclose(dapl2) < 0)
+        FAIL_STACK_ERROR;
+
+    /* Verify functionality of H5Pcopy with a dapl */
+    if ((dapl2 = H5Pcopy(dapl1)) < 0)
+        FAIL_STACK_ERROR;
+    if (H5Pget_chunk_cache(dapl2, &nslots_4, &nbytes_4, &w0_4) < 0)
+        FAIL_STACK_ERROR;
+    if ((nslots_3 != nslots_4) || (nbytes_1 != nbytes_4) || !H5_DBL_ABS_EQUAL(w0_3, w0_4))
+        FAIL_PUTS_ERROR("    Cache values from dapl2 do not match those from dapl1.");
+
+    /* Close */
+    if (H5Dclose(dsid) < 0)
+        FAIL_STACK_ERROR;
+    if (H5Sclose(sid) < 0)
+        FAIL_STACK_ERROR;
+    if (H5Pclose(fapl_local) < 0)
+        FAIL_STACK_ERROR;
+    if (H5Pclose(fapl_def) < 0)
+        FAIL_STACK_ERROR;
+    if (H5Pclose(dapl1) < 0)
+        FAIL_STACK_ERROR;
     if (H5Pclose(dapl2) < 0)
-        FAIL_STACK_ERROR /* Close dapl2, to avoid id leak */
-            if ((dapl2 = H5Dget_access_plist(dsid)) <
-                0) FAIL_STACK_ERROR if (H5Pget_chunk_cache(dapl2, &nslots_4, &nbytes_4, &w0_4) <
```

Comment:
Exactly why this kind of cleanup is nice!

Reply:
Yeah, the formatter gets super confused by some things, like tail comments after non-do-while macros w/o semicolons.